lib/usb/usb_standard.c

```diff
@@ -247,19 +248,40 @@ static int usb_standard_set_configuration(usbd_device *usbd_dev,
 					  struct usb_setup_data *req,
 					  uint8_t **buf, uint16_t *len)
 {
-	int i;
+	uint8_t i;
+	const struct usb_config_descriptor *cfg;
 
 	(void)req;
 	(void)buf;
 	(void)len;
 
-	/* Is this correct, or should we reset alternate settings. */
-	if (req->wValue == usbd_dev->current_config) {
-		return 1;
+	if(req->wValue > 0) {
+		if (req->wValue > usbd_dev->desc->bNumConfigurations) {
```

Comment:
This is currently failing the (admittedly pedantic) test case of having two configurations with bConfigurationValue of 2 and 3 (there's absolutely no requirement on those values being zero based and sequential)  I've fixed in my merge branch to scan the set of known configs.